src/watcher/watcher.worker.ts

```diff
@@ -67,6 +89,46 @@ class Parser {
         this.queue.push(...paths);
         if (!this.parsing) this.parse();
     }
+    processContent(path: string, content: string) {
+        if (this.debug)
+            console.debug(`Process Content: ${path}`)
+        let fileDetails = this.pendingContentProcessing.get(path)
+        let statBlock = this.findFirstStatBlock(content);
+        if (statBlock) {
+            if (this.debug)
+                console.debug(`StatBlocks: ${JSON.stringify(statBlock)}`)
+            const monster: Monster = Object.assign({}, YAML.parse(statBlock), {
+                note: path,
+                mtime: fileDetails.mtime
+            });
+            if (this.debug)
+                console.debug(`MONSTER: ${JSON.stringify(monster)}`)
+            this.processMonster(monster, fileDetails)
+        }
+
+        this.pendingContentProcessing.delete(path)
+        this.checkForWorkComplete();
+    }
+
+    findFirstStatBlock(content: string) : string {
+        let blocks : string[] = []
+        let parts = content.split("```")
+        parts.shift()
+        while (parts.length > 0) {
+            let codeBlock = parts.shift()
+            if (parts.shift()) {
```

Comment:
You're double shifting here; did you mean to check `if (codeBlock)` ?

Reply:
No. this is correct.  The second shift is to confirm the code block is properly terminated. In the case where you have an unterminated code block, there will have been an even number of `parts`